nodl_python/nodl/types.py

```diff
@@ -74,31 +101,40 @@ def __init__(
         message_type: str,
         publisher: bool = False,
         subscription: bool = False,
-        qos: QoSProfile = QoSPresetProfiles.SYSTEM_DEFAULT
+        qos: QoSProfile = QoSPresetProfiles.SYSTEM_DEFAULT.value
     ) -> None:
-        self.name = name
-        self.message_type = message_type
+        super().__init__(name=name, value_type=message_type)
         self.publisher = publisher
         self.subscription = subscription
 
         self.qos = qos
 
 
-class Node:
+class Node(NoDLData):
     """Data structure containing all interfaces a node exposes."""
 
     def __init__(
         self,
         *,
         name: str,
-        actions: Optional[List[Action]] = None,
-        parameters: Optional[List[Parameter]] = None,
-        services: Optional[List[Service]] = None,
-        topics: Optional[List[Topic]] = None
+        actions: List[Action] = [],
+        parameters: List[Parameter] = [],
+        services: List[Service] = [],
+        topics: List[Topic] = []
```

Comment:
Has this suddenly become safe to do in a new python that I don't know about?

Reply:
Uh it has not. Fixed 3f0f51b